### PR TITLE
feat(filters): smart episode

### PR DESCRIPTION
**Pull request by zze0s**
Add Smart episode check like autodl-irssi.

* Checks all releases for `title`, `season`, `episode` to see if it's newer or not.

Todo:

- Support date based shows like `2022-10-13`

### CODE REVIEW

File: internal/database/release.go [hunk 1]
```diff
@@ -447,3 +447,66 @@ func (repo *ReleaseRepo) Delete(ctx context.Context) error {
...
+	//		}
+	//	}
+	//}
+
+	queryBuilder := repo.db.squirrel.
+		Select("COUNT(*)").
+		From("release").
+		Where("title LIKE ?", fmt.Sprint("%", title, "%"))
+
+	if season > 0 && episode > 0 {
```
Comment (KyleSanderson) [inline internal/database/release.go:479]: These aren't quite right, they need to check the present season and episodes against that, or future seasons registered. Right now it's split brain between packs and episodes, as well as episodes failing on future seasons.

File: internal/database/release.go [hunk 2]
```diff
@@ -447,3 +447,66 @@ func (repo *ReleaseRepo) Delete(ctx context.Context) error {
...
+		queryBuilder = queryBuilder.Where(sq.And{
+			sq.GtOrEq{"season": season},
+			sq.Gt{"episode": episode},
+		})
```
Comment (KyleSanderson) [inline internal/database/release.go:483]: Maybe something like this

```
		queryBuilder = queryBuilder.Where(sq.And{
			sq.Or{
				sq.And{
					sq.Eq{"season": season},
					sq.Gt{"episode": episode},
				},
				sq.Gt{"season": season},
			}
		})
```

File: internal/database/release.go [hunk 3]
```diff
@@ -447,3 +447,66 @@ func (repo *ReleaseRepo) Delete(ctx context.Context) error {
...
+	} else if season > 0 && episode == 0 {
+		queryBuilder = queryBuilder.Where(sq.And{
+			sq.Gt{"season": season},
+			sq.Eq{"episode": 0},
+		})
+	}
```
Comment (KyleSanderson) [inline internal/database/release.go:489]: Maybe something like this on a pack.
```
		queryBuilder = queryBuilder.Where(sq.And{
			sq.Gt{"season": season}
		})
```

File: internal/database/release.go (hunk 1 above)
Comment (KyleSanderson) [reply internal/database/release.go:479]: Took a stab at it.

**Comment by zze0s** [comment]
Did some testing and looks to be working!

**Pull request merged**